src/expr/src/expr/expr_unary.rs

```diff
@@ -175,41 +189,31 @@ pub fn new_unary_expr(
                 bool_out,
             ))
         }
-        (ProstType::Not, _, _) => {
-            Box::new(UnaryNullableExpression::<BoolArray, BoolArray, _>::new(
-                child_expr,
-                return_type,
-                conjunction::not,
-            ))
-        }
-        (ProstType::IsTrue, _, _) => {
-            Box::new(UnaryNullableExpression::<BoolArray, BoolArray, _>::new(
-                child_expr,
-                return_type,
-                is_true,
-            ))
-        }
-        (ProstType::IsNotTrue, _, _) => {
-            Box::new(UnaryNullableExpression::<BoolArray, BoolArray, _>::new(
-                child_expr,
-                return_type,
-                is_not_true,
-            ))
-        }
-        (ProstType::IsFalse, _, _) => {
-            Box::new(UnaryNullableExpression::<BoolArray, BoolArray, _>::new(
-                child_expr,
-                return_type,
-                is_false,
-            ))
-        }
-        (ProstType::IsNotFalse, _, _) => {
-            Box::new(UnaryNullableExpression::<BoolArray, BoolArray, _>::new(
-                child_expr,
-                return_type,
-                is_not_false,
-            ))
-        }
+        (ProstType::Not, _, _) => Box::new(BooleanUnaryExpression::new(
+            child_expr,
+            |a| BoolArray::new(!a.data() & a.null_bitmap(), a.null_bitmap().clone()),
+            conjunction::not,
+        )),
+        (ProstType::IsTrue, _, _) => Box::new(BooleanUnaryExpression::new(
+            child_expr,
+            |a| BoolArray::new(a.to_bitmap(), Bitmap::ones(a.len())),
```

Comment:
Yes... and which code relies on this?

Reply:
For example, we do not care about the `null_bitmap` here in `IsTrue`, while not in `IsFalse`. 🥺

Reply:
IIUC, before this PR, null items always have a default value in the `data` array. But after this PR, values for null items become undefined. But this change won't affect existing code, because they don't use `raw_value` or `raw_iter` API.

Reply:
Oops. I missed that `to_bitmap` is actually `data & null_bitmap`. 🥵

Reply:
Oh I see what you mean.

`a.to_bitmap()` equals to `a.data() & a.null_bitmap()`. So we do care about the `null_bitmap` here.🥺